src/ecpool.erl

```diff
@@ -81,13 +81,35 @@ pool_spec(ChildId, Pool, Mod, Opts) ->
 %% @doc Start the pool sup.
 -spec(start_pool(pool_name(), atom(), [option()]) -> {ok, pid()} | {error, term()}).
 start_pool(Pool, Mod, Opts) ->
-    ecpool_pool_sup:start_link(Pool, Mod, Opts).
+    %% See start_sup_pool/3 for an explanation of InitialConnectResponseRef
+    InitialConnectResponseRef = make_ref(),
+    {ok, Pid} = OkResponse = ecpool_pool_sup:start_link(Pool, Mod, Opts, {self(), InitialConnectResponseRef}),
+    case aggregate_initial_connect_responses(InitialConnectResponseRef, Opts) of
+        ok ->
+           OkResponse ;
+        Error ->
+            exit(Pid, normal),
```

Comment:
isn't `normal` ignored if the process is trapping exit?

Reply:
Thanks, I have to look this up. The logic should work anyway since the pool is closed in the ecpool module when an error is returned. (I'm assuming that the ecpool module is the public interface for the library.)

Reply:
it's a supervisor, maybe call a regular `stop` api?

Reply:
Does the supervisor have a `stop` API? There is no stop function in the supervisor module: https://www.erlang.org/doc/apps/stdlib/supervisor.html

Reply:
right, there is no stop api for supervisor, only apis to stop/terminate its children

Reply:
There are `gen_server:stop/1,3` / `gen:stop/1,3` that should be usable for that for the foreseeable future, but strictly speaking this would create "relying on the implementation detail" type of problem.

Reply:
I have confirmed that sending the exist signal with the reason normal has the desired effect (the supervisor and its children are terminated). Sending and exit signal with reason shutdown does the same except that the calling process also get the exit signal (as it is linked). It seems like the supervisor ignores the reason for the signal and handles all signals the same way:

https://github.com/erlang/otp/blob/master/lib/stdlib/src/supervisor.erl#L1176

I have changed the code to use (see [5f5dfcf](https://github.com/emqx/ecpool/pull/50/commits/5f5dfcff0bef28c31e4741d2874379a83b83c4e5)):

```
            unlink(Pid),
            exit(Pid, shutdown),
```

just because using the the shutdown reason seems to be a more common way to shut down a supervisor.